File: proj-euler/P12.py
"""
the n^th tri. num is the sum of the first n natural numbers
find the first one that has >500 factors (ie, 6 has 4 factors 1236).
"""

def main():
	number=0
	swapped = True
	while (swapped):
		number=number+1
		if (findFactorCount(triNum(number))>500):
			print triNum(number)
			swapped=False	

def triNum(count):
	#gives the *count*^th triangular number ...
	return (count*(count+1))/2

def findFactorCount(number):
	count=0
	#insert a particular triangular number to see its number of factors
	for x in range(number):
		if (number%(x+1)==0):
			count+=1
	return count	

main()

File: proj-euler/P12.java
/*
find the first triangular number with >500 factors
*/

public class P12 {

	public static void main(String argv[]){
	    boolean switc = true;		
	    for (int i=1; switc; i++) {		
		int result = findFact(triNum(i));
		//if (i<50) System.out.println(""+result+" "+triNum(i));
	    	if (result>500) {
		        System.out.println(""+result+" "+triNum(i));
			switc = false;
		}
	    }
	}

	public static int triNum(int count){
		//return the "count^th" triangular number
		return (count*(count+1))/2;
	}

	public static int findFact(int num) {
		int count=0; //all numbers have at least themselves as a factor
			     //also have 1 as factor...
		int divisor = 1;
		for (int i=1;i<(num/divisor);i++) {
			if (num%i==0) {
			   if (i*i==num) count+=1;
		           else count+=2;
			   divisor=i;
			}
		}
		return count;
	}

}
